### PR TITLE
More methods extracted in BindingTouch [Generator]

### PR DESCRIPTION
**Pull request by dustin-wojciechowski**
Extracted a good amount of code from Main3 which should make it easier to read. Tried to keep changes minimal.

### CODE REVIEW

File: src/bgen/BindingTouch.cs [hunk 1]
```diff
@@ -334,13 +317,49 @@ int Main3 (string [] args)
...
+			api = TypeManager.ParseApi (apiAssembly, config.ProcessEnums);
 			namespaceCache ??= new NamespaceCache (
 				CurrentPlatform,
 				config.HelperClassNamespace ?? firstApiDefinitionName,
 				LibraryManager.DetermineSkipSystemDrawing (LibraryInfo.TargetFramework)
 			);
 
+
+		} catch (Exception ex) {
+			ErrorHelper.Show (ex);
```
Comment (rolfbjarne) [inline src/bgen/BindingTouch.cs:329]:
```suggestion
			ErrorHelper.Show (ex);
			return false;
```

File: src/bgen/BindingTouch.cs [hunk 2]
```diff
@@ -137,123 +137,113 @@ static int Main2 (string [] args)
...
-			return 0;
+			return false;
 		}
 
-		libraryInfo = LibraryInfo.LibraryInfoBuilder.Build (references, config);
-		CurrentPlatform = LibraryManager.DetermineCurrentPlatform (TargetFramework.Platform);
+		return true;
+	}
 
+	public bool TryInitializeApi (BindingTouchConfig config, out Api? api)
```
Comment (rolfbjarne) [inline src/bgen/BindingTouch.cs:244]:
```suggestion
	public bool TryInitializeApi (BindingTouchConfig config, [NotNullWhen (true)] out Api? api)
```

File: src/bgen/BindingTouch.cs [hunk 3]
```diff
@@ -334,13 +317,49 @@ int Main3 (string [] args)
...
+			return 1;
+
+		return 0;
+	}
+
+	bool TryGenerate (BindingTouchConfig config, Api? api)
+	{
+		if (api is null)
+			return false;
+
```
Comment (rolfbjarne) [inline src/bgen/BindingTouch.cs:361]: With the [NotNullWhen (true)] attribute in TryInitializeApi, the Api parameter doesn't need to be nullable anymore:

```suggestion
	bool TryGenerate (BindingTouchConfig config, Api api)
	{
```

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by dustin-wojciechowski** [comment]
Test Failures unrelated, see https://github.com/xamarin/maccore/issues/2488

**Pull request merged**